bson/primitive/decimal.go

```diff
@@ -25,60 +39,61 @@ func NewDecimal128(h, l uint64) Decimal128 {
 	return Decimal128{h: h, l: l}
 }
 
-// GetBytes returns the underlying bytes of the BSON decimal value as two uint16 values. The first
+// GetBytes returns the underlying bytes of the BSON decimal value as two uint64 values. The first
 // contains the most first 8 bytes of the value and the second contains the latter.
 func (d Decimal128) GetBytes() (uint64, uint64) {
 	return d.h, d.l
 }
 
 // String returns a string representation of the decimal value.
 func (d Decimal128) String() string {
-	var pos int     // positive sign
-	var e int       // exponent
-	var h, l uint64 // significand high/low
+	var posSign int      // positive sign
```

Comment:
I'd suggest making this a `bool` as well to be consistent with the other declarations of `posSign`.

I notice that it is used to do a slice of some strings. I'd make this follow the way you handled `ErrParseNaN`, `ErrParseInf` and etc., by making constants (in this case, you can indeed make `const`).